### ISSUE TITLE
How to change batch size?

### ISSUE THREAD

**Issue by Pingcheng-Jian** [lucidrains/robotic-transformer-pytorch#9]
The default batch size of this model is 2. I want to use a larger batch size such as 32, but get the error:
`classifier_free_guidance_pytorch/classifier_free_guidance_pytorch.py", line 394, in forward
    return hiddens * (scale + 1) + shift
RuntimeError: The size of tensor a (320) must match the size of tensor b (20) at non-singleton dimension 0`

How to solve it?

**Comment by lucidrains** [comment]
@Pingcheng-Jian could you show me the full script where it errors? is your text a list of strings of size 32 as well?

**Comment by Pingcheng-Jian** [comment]
`import torch
from robotic_transformer_pytorch import MaxViT, RT1

vit = MaxViT(
    num_classes = 1000,
    dim_conv_stem = 64,
    dim = 96,
    dim_head = 32,
    depth = (2, 2, 5, 2),  # (2, 2, 5, 2),
    window_size = 7,
    mbconv_expansion_rate = 4,
    mbconv_shrinkage_rate = 0.25,
    dropout = 0.1
)

model = RT1(
    vit = vit,
    num_actions = 7,  # originally 11
    depth = 6,
    heads = 8,
    dim_head = 64,
    cond_drop_prob = 0.2
)

video = torch.randn(32, 3, 10, 224, 224)  # b c f h w

instructions = [
    'bring me that apple sitting on the table',
    'please pass the butter'
]

train_logits = model(video, instructions) # (2, 6, 11, 256) # (batch, frames, actions, bins)
print(train_logits)
print(train_logits.shape)
print(train_logits[0][0][0])

model.eval()
eval_logits = model(video, instructions, cond_scale = 3.) # classifier free guidance with conditional scale of 3
print(eval_logits)
print(eval_logits.shape)
print(eval_logits[0][0][0])`

This is the code. I just modify the video from torch.randn(2, 3, 6, 224, 224) to torch.randn(32, 3, 10, 224, 224). And I get error:
RuntimeError: The size of tensor a (320) must match the size of tensor b (20) at non-singleton dimension 0

**Comment by Pingcheng-Jian** [comment]
Also, if I change the line `torch.randn(32, 3, 10, 224, 224)` back to `torch.randn(2, 3, 10, 224, 224)`, it can run without error.

**Comment by lucidrains** [comment]
@Pingcheng-Jian want to try 0.2.2 and see if the error message is more informative?

**Comment by lucidrains** [comment]
@Pingcheng-Jian want to try 0.2.2 and see if the error message is more informative?

**Comment by Pingcheng-Jian** [comment]
Ok, Now I got this error:
`robotic_transformer_pytorch/robotic_transformer_pytorch.py", line 614, in forward
    assert num_texts == video.shape[0], f'you only passed in {num_texts} strings for guiding the robot actions, but received batch size of {video.shape[0]} videos'
AssertionError: you only passed in 2 strings for guiding the robot actions, but received batch size of 32 videos`

I guess I need to pass in 32 sentences instead of 2.

**Comment by Pingcheng-Jian** [comment]
Yes, so after I use the version 0.2.2, and use the `instructions = ['please pass the butter'] * 32`, this testing script works. However, a simple forward pass of this 32 batch takes about 1 minute. Such a long foward passing time can make the training take for ever.

**Comment by lucidrains** [comment]
are you on a gpu?

**Comment by lucidrains** [comment]
@Pingcheng-Jian yes no problem

let me know if you get any interesting results

plan on doing more open sourcing in robotics space soon
